doc/guides/modules.rst

```diff
@@ -10,14 +10,342 @@ system those are called *modules*. These modules must be integrated with the
+
+* Modules should use "zephyr" as the default name for the repository main
+  branch. Branches for specific purposes, for example, a module branch for
+  an LTS Zephyr version, shall have names starting with the 'zephyr\_' prefix.
+
+  .. note::
+
+     It is **not** recommended in module repositories to have a 'master'
+     branch mirroring the master branch of the external repository.
+
+.. _modules_synchronization:
+
+Synchronizing with upstream
+===========================
+
+Requirements for allowed practices
+----------------------------------
+
+Any changes to module repositories, as a result of synchronizing with,
+upstream shall be brought in by pull requests. These pull requests
+shall be merg-able (i.e. with Rebase & Merge option, using Github UI).
+This ensures that a linear history is preserved in the module repositories.
+It also allows to easily review the incoming changea and run Zephyr CI,
+git lint, identity, and license checks directly on the set of changes
+that are to be brought into the module.
+
+.. note::
+
+     Force-pushing to a module's main branch is not allowed.
+
+Allowed practices
+---------------------
+
+The following practices are allowed and should be followed in all modules.
+
+**Updating modules with a diff from upstream:**
+Upstream changes brought as a single commit (manual diff) in a PR against
+the module's main branch. This approach is simple and should be applicable
+to all modules with the downside of supressing the upstream history in the
+module repository.
+
+The commit message is expected to identify the upstream repository, the
+version to which the module is updated (upstream version, tag, commit,
+etc.), and the reason for the doing the update.
+
+**Cherry picking upstream commits:**
+Upstream changes brought in by cherry-picking the upstream commits and
+submitting a PR with these commits against the module main branch. This
+approach is applicable to modules with an upstream project Git repository.
+The main advantages of this approach are the following
+
+* the upstream repository history may be preserved in the module repository,
+  if the cherry picked commits have a pointer to the original upstream commit
+* downstream module patches are less likely to be unintentionally overwritten
+  by the synchronization process.
+
+
+Contributing to Zephyr modules
+******************************
+
+.. _modules_contributing:
+
+
+Individual Roles & Responsibilities
+===================================
+
+To facilitate management of Zephyr module repositories, the following
+individual roles are defined.
+
+**Administration:** Each Zephyr module shall have an administrator
+who is responsible for managing access to the module repository,
+for example, for adding individuals as Collaborators in the repository
+at the request of the module code owner. Module administrators are
+members of the Administrators' team, that is a group of project
+members with admin rights to Module Github repositories.
```

Comment:
For now it will be the zephyr-project-rtos administrators , but the policy is more flexible - it can be any other group of people with admin right on module repositories.
We need a team to administer the module repos, that's all.